python/ground_git.py

```diff
@@ -0,0 +1,1056 @@
+
+    """
+    def commit(self):
+        stage = []
+        for kee in self.graph.ids:
+            if kee in self.graph.nodes:
+                serial = self.graph.nodes[kee].to_json()
+            elif kee in self.graph.nodeVersions:
+                serial = self.graph.nodeVersions[kee].to_json()
+            elif kee in self.graph.edges:
+                serial = self.graph.edges[kee].to_json()
+            elif kee in self.graph.edgeVersions:
+                serial = self.graph.edgeVersions[kee].to_json()
+            elif kee in self.graph.graphs:
+                serial = self.graph.graphs[kee].to_json()
+            elif kee in self.graph.graphVersions:
+                serial = self.graph.graphVersions[kee].to_json()
+            elif kee in self.graph.structures:
+                serial = self.graph.structures[kee].to_json()
+            elif kee in self.graph.structureVersions:
+                serial = self.graph.structureVersions[kee].to_json()
+            elif kee in self.graph.lineageEdges:
+                serial = self.graph.lineageEdges[kee].to_json()
+            elif kee in self.graph.lineageEdgeVersions:
+                serial = self.graph.lineageEdgeVersions[kee].to_json()
+            elif kee in self.graph.lineageGraphs:
+                serial = self.graph.lineageGraphs[kee].to_json()
+            else:
+                serial = self.graph.lineageGraphVersions[kee].to_json()
+            assert serial is not None
+            with open(str(kee) + '.json', 'w') as f:
+                f.write(serial)
+            stage.append(str(kee) + '.json')
+        repo = git.Repo.init(os.getcwd())
+        repo.index.add(stage)
+        repo.index.commit("ground commit")
+        tree = repo.tree()
+        with open('.jarvis', 'w') as f:
+            for obj in tree:
+                commithash = self.__run_proc__("git log " + obj.path).replace('\n', ' ').split()[1]
+                if obj.path != '.jarvis':
+                    f.write(obj.path + " " + commithash + "\n")
+        repo.index.add(['.jarvis'])
+        repo.index.commit('.jarvis commit')
+
+    def load(self):
+        if self.graph.ids:
+            return
+        os.chdir('../')
+
+        def is_number(s):
+            try:
+                float(s)
+                return True
+            except ValueError:
+                return False
+
+        listdir = [x for x in filter(is_number, os.listdir())]
+
+        prevDir = str(len(listdir) - 1)
+        os.chdir(prevDir)
+        for _, _, filenames in os.walk('.'):
+            for filename in filenames:
+                filename = filename.split('.')
+                if filename[-1] == 'json':
+                    filename = '.'.join(filename)
+                    with open(filename, 'r') as f:
+                        self.to_class(json.loads(f.read()))
+        os.chdir('../' + str(int(prevDir) + 1))
+    """
+
+class GroundImplementation(GroundAPI):
```

Comment:
What is this?

Reply:
Holdover from the file in jarvis, unnecessary and will delete